### DISCUSSION TITLE
cannot find module 'ext.slugs' (No module named 'ext')

### DISCUSSION THREAD

**Discussion by pitosalas** [squidfunk/mkdocs-material#7242] [Q&A] [answered]
I am following the doc to create an extension to change the slug format. I created the ext/ directory with the __init__.py. And still I get this error:

`~/mydev/wpconvert (main) mkdocs build -v
DEBUG   -  Loading configuration file: /Users/pitosalas/mydev/wpconvert/mkdocs.yml
Error: MkDocs encountered an error parsing the configuration file: while constructing a Python object
cannot find module 'ext.slugs' (No module named 'ext')
  in "/Users/pitosalas/mydev/wpconvert/mkdocs.yml", line 30, column 21`

**Comment by kamilkrzyskow** [comment, marked as answer]
Hello @pitosalas,
you can run:
```shell
python -c "import sys; print(*sys.path, sep='\n')"
```
to check what directories are in your python interpreter path.

You can add your current directory with `export PYTHONPATH=.` on a UNIX-like terminal.
Note that `.` means every "current directory" at any given path, so watch out for name conflicts, but overall didn't happen to me yet despite working in many MkDocs projects. 
You can also add an absolute path.

**Comment by pitosalas** [comment]
Thanks, it worked! But I always thought that . was the first place where python looks? Can you help me understand? Thanks!

**Reply by kamilkrzyskow** [reply]
> But I always thought that . was the first place where python looks? Can you help me understand?

If you use Python to open a file `open("some.json", encoding="utf-8")` then it will look in your current working directory, so by default the same path you run `python script.py`, so it will open `./some.json`. Locality takes priority and is the default.

However, this applies to normal file system operations, in this case with `PYTHONPATH=.` we're setting the current directory (any "current directory") to be a root of a lib directory, which is used for package lookup during `import`.

Let's say that instead of naming your directory `ext` you'd call it `json`, also with an `__init__.py` inside.

Now running a script like this:
```py
import json

obj = json.loads('{ "jsonData": "somedata" }')

print(obj)
```
would crash Python with `AttributeError: module 'json' has no attribute 'loads'`

This happens because the `PYTHONPATH=.` is looked up first and `json` module from your directory is loaded.
That's why some caution needs to be in place when it comes to setting any "current directory" as a package root.

**Comment by pitosalas** [comment]
And if I can add a question: I am building my site in a github action. How would I achieve the same thing there?

Thanks!

**Reply by kamilkrzyskow** [reply]
Assuming that `PYTHONPATH=.` is enough to make it work on the linux runners, then you can modify your `mkdocs` command in your `CI.yml` workflow file and prepend it with `export PYTHONPATH=. ; mkdocs...`
